packages/data-service/src/cancellable-promise.ts

```diff
@@ -0,0 +1,73 @@
+export const OPERATION_CANCELLED_ERROR = 'OperationCancelledError';
+
+const OPERATION_CANCELLED_MESSAGE = 'The operation was cancelled.';
+
+class PromiseCancelledError extends Error {
+  name = OPERATION_CANCELLED_ERROR;
+}
+
+export const createCancelPromiseError = (): Error => {
+  return new PromiseCancelledError(OPERATION_CANCELLED_MESSAGE);
+};
```

Comment:
This is a bit nicer:

```suggestion
class PromiseCancelledError extends Error {
  name = OPERATION_CANCELLED_ERROR;
  
  constructor() {
     super(OPERATION_CANCELLED_MESSAGE);
  }
}
```

Like that we don't need `createCancelPromiseError` anymore and just call `new PromiseCancelledError()` everywhere.